src/middleware/index.ts

```diff
@@ -46,11 +46,22 @@ export const logErrors = (
   next(err);
 };
 
+
+export class ResponseErr extends Error {
+  statusCode: number
+  constructor(msg: string, statusCode = 500) {
+    super(msg);
+    this.statusCode = statusCode;
+  }
+}
+
```

Comment:
(Unrelated): I noticed that most of error responses are marked with a `500` status code but most of the time it should be `400` for bad requests. The point is to distinguish between different error responses. I created this class so that when throwing the error we can specify the **status code** and `errorHandler` middleware will send the right status code to the client.